mesonbuild/ast/interpreter.py

```diff
@@ -85,11 +87,27 @@ class MockRunTarget(MesonInterpreterObject):
 _V = T.TypeVar('_V')
 
 
+@dataclasses.dataclass
+class GlobalState(_GlobalState):
+
+    subproject_dir: str
+
+
+class State(_State):
```

Comment:
`State` is not a great name because it is such a common word. A more descriptive name would be better